patches/tock/12-connect-vendor-hid-usb-interface.patch

```diff
@@ -0,0 +1,516 @@
+diff --git a/capsules/src/usb/usb_ctap.rs b/capsules/src/usb/usb_ctap.rs
+index 2c1ecf934..2c91c0968 100644
+--- a/capsules/src/usb/usb_ctap.rs
++++ b/capsules/src/usb/usb_ctap.rs
+@@ -32,8 +32,7 @@ pub trait CtapUsbClient {
+     fn can_receive_packet(&self, app: &Option<&mut App>) -> bool;
+ 
+     // Signal to the client that a packet has been received.
+-    // If App is not supplied, it will be found from the implemntation's members.
+-    fn packet_received(&self, packet: &[u8; 64], app: Option<&mut App>);
++    fn packet_received(&self, packet: &[u8; 64], endpoint: usize, app: Option<&mut App>);
+ 
+     // Signal to the client that a packet has been transmitted.
+     fn packet_transmitted(&self);
+@@ -49,7 +48,7 @@ impl<'a, 'b, C: hil::usb::UsbController<'a>> CtapUsbSyscallDriver<'a, 'b, C> {
+         CtapUsbSyscallDriver { usb_client, apps }
+     }
+ 
+-    fn app_packet_received(&self, packet: &[u8; 64], app: &mut App) {
++    fn app_packet_received(&self, packet: &[u8; 64], endpoint: usize, app: &mut App) {
+         if app.connected && app.waiting && app.side.map_or(false, |side| side.can_receive()) {
+             if let Some(buf) = &mut app.buffer {
+                 // Copy the packet to the app's allowed buffer.
+@@ -57,7 +56,7 @@ impl<'a, 'b, C: hil::usb::UsbController<'a>> CtapUsbSyscallDriver<'a, 'b, C> {
+                 app.waiting = false;
+                 // Signal to the app that a packet is ready.
+                 app.callback
+-                    .map(|mut cb| cb.schedule(CTAP_CALLBACK_RECEIVED, 1, 0));
```

Comment:
You can delete patch 11, it was supposed to be superseded by this PR, and its only change is overwritten here.

Reply:
Done.